tests/test.c

```diff
@@ -83,98 +75,66 @@ static void fprintf_delimiter(FILE *stream, char *format, ...) {
     // Print string
     fprintf(stream, " %s ", str);
     chars_remaining -= strlen(str) + num_spaces;
+    
     // Print stars on right
     for (int i = 0; i < chars_remaining; i++) {
         fprintf(stream, "*");
     }
     fprintf(stream, "\n");
 }
 
-// creates a child process that duplicates the output from the test to both standard out and to a temp file
-void start_test (char *test_name) {
-    fprintf_delimiter(stdout, "Starting test: \"%s\"", test_name);
-    fflush(stdout);
+// ********************************************** PUBLIC TEST FRAMEWORK FUNCTIONS ************************************** //
 
+// creates a pipe to route stdout and stdin to for output handling, spawns the output handler thread
+void start_test (char *test_description) {
+    printf("************************************** Starting test: \"%s\" **************************************\n", test_description);
+    fflush(stdout);
+    
     // save the test name
-    global_test_name = malloc(strlen(test_name) + 1);
-    strcpy(global_test_name, test_name);
-
+    global_test_name = malloc(strlen(test_description) + 1);
+    strcpy(global_test_name, test_description);
+    
     // create a pipe
     if (pipe(pipe_fd) < 0) {
         printf("pipe: %s\n", strerror(errno));
     }
-
-    // fork this process; in the child, route stdin to read end of pipe; in parent, route stdout to write end of pipe
-    if ((output_redirect = fork()) < 0) {
-        printf("fork: %s\n", strerror(errno));
-    } else if (output_redirect == 0) { // child
-        // open the temp file
-        if ((temp_fp = fopen(TEMP_FILE, "w")) == NULL) {
-            printf("open: cannot create temp file: %s\n", strerror(errno));
-        }
-
-        // route stdin to read end of pipe
-        if (dup2(pipe_fd[0], fileno(stdin)) == -1) {
-            fprintf(stderr, "dup2 stdin to read end of pipe: %s\n", strerror(errno));
-        }
-        close(pipe_fd[1]); // don't need write end
-
-        child_main_function();
-    } else { // parent
-        // save the standard out attached to the terminal for later
-        save_std_out = dup(fileno(stdout));
-
-        // route stdout to write end of pipe
-        if (dup2(pipe_fd[1], fileno(stdout)) == -1) {
-            fprintf(stderr, "dup2 stdout to write end of pipe: %s\n", strerror(errno));
-        }
-        close(pipe_fd[0]); // don't need read end
-
-        // kill the child on Ctrl-C (SIGINT)
-        signal(SIGINT, kill_child);
-
-        // register the clean up handler
-        atexit(cleanup_handler);
+    // save the standard out attached to the terminal for later
+    save_std_out = dup(fileno(stdout));
+    
+    // route stdin to read end of pipe
```

Comment:
comment should be other way around. read end gets routed to stdin

Reply:
👌